src/vaults/vaults.service.ts

```diff
@@ -33,6 +52,103 @@ export class VaultsService {
     return VaultsService.loadVault(chain, vaultDefinition, currency);
   }
 
+  async listVaultHarvests(chain: Chain): Promise<VaultHarvestsMap> {
+    const harvestsWithSnapshots = await Promise.all(
+      chain.vaults.map(async (vault) => {
+        return {
+          vault: vault.vaultToken,
+          harvests: await this.getVaultHarvests(chain, vault.vaultToken),
+        };
+      }),
+    );
+
+    return harvestsWithSnapshots.reduce((acc, harvestWithSnapshot) => {
+      acc[harvestWithSnapshot.vault] = harvestWithSnapshot.harvests;
+      return acc;
+    }, <VaultHarvestsMap>{});
+  }
+
+  async getVaultHarvests(chain: Chain, vaultAddr: VaultDefinition['vaultToken']): Promise<VaultHarvestsExtended[]> {
+    const vaultHarvests: VaultHarvestsExtended[] = [];
+
+    vaultAddr = ethers.utils.getAddress(vaultAddr.toLowerCase());
```

Comment:
we don't need to use `toLowerCase()` because `getAddress` converts the value to a checksum addr which is anyway not relying on lowercase

Reply:
done;